torchgeo/datasets/dior.py

```diff
@@ -0,0 +1,392 @@
+            split: split of the dataset to use, one of 'train', 'val', 'test'
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            AssertionError: If *split* argumnet is invalid.
+        """
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+        self.download = download
+
+        assert split in self.valid_splits, f'Split must be one of {self.valid_splits}.'
+        self.split = split
+
+        self._verify()
+
+        self.sample_df = pd.read_parquet(os.path.join(self.root, 'sample_df.parquet'))
+
+        self.sample_df = self.sample_df[
+            self.sample_df['split'] == self.split
+        ].reset_index(drop=True)
+
+        self.class_to_idx: dict[str, int] = {c: i for i, c in enumerate(self.classes)}
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.sample_df)
+
+    def __getitem__(self, idx: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            idx: index to return
+
+        Returns:
+            data and label at that index
+        """
+        row = self.sample_df.iloc[idx]
+
+        image = self._load_image(os.path.join(self.root, row['image_path']))
+
+        sample: dict[str, Tensor] = {'image': image}
+
+        if self.split != 'test':
+            boxes, labels = self._load_target(
+                os.path.join(self.root, row['label_path'])
+            )
+            sample['bbox_xyxy'] = boxes
+            sample['label'] = labels
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        with Image.open(path) as img:
+            array: np.typing.NDArray[np.int_] = np.array(img.convert('RGB'))
+            tensor: Tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: Path) -> tuple[Tensor, Tensor]:
+        """Load the target mask for a single image.
+
+        Args:
+            path: path to the annotation file
+
+        Returns:
+            the target bounding boxes and labels
+        """
+        parsed = parse_pascal_voc(path)
+        boxes = torch.tensor(parsed['bboxes'], dtype=torch.float32)
```

Comment:
Are the boxes pixel-based? Does float make sense here?

Reply:
Yeah they are float values